### PR TITLE
956 update subjects v1

### PR DESCRIPTION
**Pull request by goodviber**
### Context
To improve the selection of subjects and make it clear what bursary and scholarship information is available, we have updated the subject page.
What needs to be done

change primary subjects to a list of radios
remove the details component for secondary subjects
move second subject drop down to below first subject drop down
move bursary and scholarship content to top of page
update bursary and scholarship content

### Changes proposed in this pull request
Split subjects view into partials
### Guidance to review
create new course and edit existing courses
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/decorators/course_decorator.rb [hunk 1]
```diff
@@ -280,6 +280,7 @@ def further_education?
     level == "further_education" && subjects.any? { |s| s.subject_name == "Further education" || s.subject_code = "41" }
   end
 
+  # TODO: check we still need this
```
Comment (defong) [inline app/decorators/course_decorator.rb:283]: Do we need it then?

File: app/decorators/course_decorator.rb [hunk 2]
```diff
@@ -291,12 +292,13 @@ def subject_page_title
     end
   end
 
+  # TODO: check we still need this
```
Comment (defong) [inline app/decorators/course_decorator.rb:295]: Do we need it then?

File: app/views/publish/courses/subjects/_primary_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<%= render "publish/shared/error_wrapper", error_keys: [:master_subject_id], data_qa: "course__master_subject_id" do %>
+  <fieldset class="govuk-fieldset">
+    <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+      <h1 class="govuk-fieldset__heading" data-qa="page-heading">
+        <% if course.course_code %>
+          <%= render CaptionText.new(text: course.name_and_code) %>
+        <% end %>
+
+        <%#=  Change this ? course.subject_page_title %>
```
Comment (defong) [inline app/views/publish/courses/subjects/_primary_form.html.erb:9]: Leftovers?

File: app/views/publish/courses/subjects/_secondary_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<%= render "publish/shared/error_wrapper", error_keys: [:subjects], data_qa: "course__subjects" do %>
+  <fieldset class="govuk-fieldset">
+    <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+      <h1 class="govuk-fieldset__heading" data-qa="page-heading">
+        <% if course.course_code %>
+          <%= render CaptionText.new(text: course.name_and_code) %>
+        <% end %>
+
+        <%#= Change this ? course.subject_page_title %>
```
Comment (defong) [inline app/views/publish/courses/subjects/_secondary_form.html.erb:9]: Leftovers?

File: spec/decorators/course_decorator_spec.rb [hunk 5]
```diff
@@ -823,7 +823,7 @@
       let(:course) { build_stubbed(:course, level: "secondary") }
 
       it "returns the correct input label" do
-        expect(subject_input_label).to eq("Secondary subject")
+        expect(subject_input_label).to eq("First subject")
```
Comment (defong) [inline spec/decorators/course_decorator_spec.rb:826]: Where is the second one?

File: app/views/publish/courses/subjects/_secondary_form.html.erb (hunk 4 above)
Comment (goodviber) [reply app/views/publish/courses/subjects/_secondary_form.html.erb]: Gone

File: app/views/publish/courses/subjects/_primary_form.html.erb (hunk 3 above)
Comment (goodviber) [reply app/views/publish/courses/subjects/_primary_form.html.erb]: Gone

File: app/decorators/course_decorator.rb (hunk 2 above)
Comment (goodviber) [reply app/decorators/course_decorator.rb]: No

File: app/decorators/course_decorator.rb (hunk 1 above)
Comment (goodviber) [reply app/decorators/course_decorator.rb:283]: Yes

File: spec/decorators/course_decorator_spec.rb (hunk 5 above)
Comment (goodviber) [reply spec/decorators/course_decorator_spec.rb]: Gone

File: app/helpers/publish/subject_helper.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        PrimarySubjectInput.new(1 + subject.subject_code.to_i, subject.subject_name)
+      end
+    end
+
+  private
+
+    PrimarySubjectInput = Struct.new(:code, :name)
+
+    def primary_subjects
+      Subject.where(type: 'PrimarySubject')
```
Comment (defong) [inline app/helpers/publish/subject_helper.rb:16]:
```suggestion
      PrimarySubject
```

**Review by defong — APPROVED** [review]

**Pull request merged**